File: Problems/ContainsDuplicate/contains_duplicate.go
/* 

*/
package main

import (
	"fmt"
	"encoding/json"
    "io/ioutil"
	"log"
)

type TestCase struct {
	Input []int
	Expected bool
	Comment string
}
type TestCases struct {
	Tests []TestCase
}


func containsDuplicate(input []int) bool {
	var duplicates = map[int]int{}

	for _, value := range input { 
		duplicates[value] += 1

		if duplicates[value] > 1 {
			return true
		}
	}

	return false
}

func main() {
	content, err := ioutil.ReadFile("./test_cases.json")
    if err != nil {
        log.Fatal("Error when opening file: ", err)
    }

	var testCases TestCases
    err = json.Unmarshal(content, &testCases)
    if err != nil {
        log.Fatal("Error during Unmarshal(): ", err)
		
    }

	for _, test := range testCases.Tests {
		var input = test.Input
		var expected = test.Expected
		var result = containsDuplicate(input)
		
		var passed = result == expected
		fmt.Printf("Input: %v\nResult: %v\nExpected: %v\nPassed: %v\n\n", input, result, expected, passed)
	}

}

File: Problems/ContainsDuplicate/contains_duplicate.py
import json
from typing import List

def contains_duplicate(input:List[int])->bool:
    duplicates = set()
    for i in input:
        if i in duplicates:
            return True

        duplicates.add(i)

    return False

if __name__ == "__main__":
    with open("test_cases.json", mode='r', encoding="utf-8-sig") as file:
        text = file.read() 
        test_cases = json.loads(text)
        for test in test_cases['tests']:
            input = test['input']
            expected = test['expected']
            result = contains_duplicate(input=input)
            passed = expected == result
            print(f"Input:{input}\nResult: {result}\nExpected: {expected}\nPassed: {passed}\n\n")